react_week_1.md

```diff
@@ -0,0 +1,52 @@
+# Week 1
+
+## React. Components
+
+Should be done:
+1.	Create separate branch for this task.
+2.	Use CRA with template --typescript.
+3.	Set up eslint and prettier so that project would not build if having errors. Use [configs](https://github.com/rolling-scopes-school/tasks/blob/master/react/modules/module01/configs.md) [RU]
+4.	Add React-Router version 6. Add header showing current page, also add pages “About Us”, “404”. If trying to input unknown route into url – application should redirect to “404”.
+5.	Draw the following on the main page:
+* Search Bar.\
+examples:\
+https://studio.uxpincdn.com/studio/wp-content/uploads/2020/09/BlogHeader_SearchBar_1200x600.png \
+https://www.sliderrevolution.com/wp-content/uploads/2021/02/cssheader1.jpg \
+Input value should be saved to LocalStorage during component’s unmount. While initializing look at it's value in LocalStorage and show it.
+* Cards. The more details are on the card – the better.\
+examples: \
+https://www.webdesignerdepot.com/cdn-origin/uploads/2017/01/behance.jpg \
+https://www.webdesignerdepot.com/cdn-origin/uploads/2017/01/rightmove.jpg \
+https://www.webdesignerdepot.com/cdn-origin/uploads/2017/01/awwwards.jpg \
+**In the situation when the access to life cycles of a component is needed or there is a necessity of state – use class components. Using hooks is forbidden at this stage.**
+6.	Add tests for components. Mock LocalStorage. Add test for one card, as well as tests for the list of all cards. Usage of Testing Library React or Enzyme is allowed, but only one of them. Preferred Testing Library React.
+
+All logical parts should be set into separate components.
+### Score
+The task will be assessed by mentor in march. For checking convenience create Pull Request to main(master) branch (**DO NOT MERGE**). 
+1.	Tests – **4 points**
+2.	eslint, prettier – **2 points**
+3.	React App + components and router – **9 points**
+
+If one of the mentioned above list’s items is not accomplished – grade for the task is **0 points**. If there is direct DOM manipulations – **0 points**.
```

Comment:
If one of the mentioned above list’s items is not accomplished, the task grade is **0 points**. If there are direct DOM manipulations – **0 points**.